src/zarr/group.py

```diff
@@ -181,10 +183,10 @@ async def open(
             assert zarr_json_bytes is not None
             group_metadata = json.loads(zarr_json_bytes.to_bytes())
 
-        return cls.from_dict(store_path, group_metadata)
+        return await cls.from_dict(store_path, group_metadata)
 
     @classmethod
-    def from_dict(
+    async def from_dict(
```

Comment:
This is a notable change that was needed to get the hierarchy API to work. Previously, `from_dict` was not async, but it should be.